syntax = "proto3";

package protobufs.common.v1;

import "protobufs/common/v1/language.proto";

message Problem {
    string id = 1;
    string title = 2;
    string description = 3;
    repeated TestData tests = 4;
    repeated common.v1.ProgrammingLanguage supported_languages = 5;

    message TestData {
        string id = 1;
        optional string arguments = 2;
        repeated TestFile input_files = 3;
        optional string expected_stdout = 4;
        repeated TestFile expected_files = 5;
    }

    message TestFile {
        string path = 1;
        bytes data = 2;
    }
}

message ProblemSummary {
    string id = 1;
    string title = 2;
    string summary = 3;
}
